third_party/dependency_analyzer/src/test/io/bazel/rulesscala/dependencyanalyzer/AstUsedJarFinderTest.scala

```diff
@@ -318,4 +373,60 @@ class AstUsedJarFinderTest extends FunSuite {
       )
     }
   }
+
+  test("java interface field and method is direct") {
+    withSandbox { sandbox =>
+      sandbox.compileJava(
+        className = "A",
+        code = "public interface A { int a = 42; }"
+      )
+      val bCode =
+        """
+          |class B {
+          |  def foo(x: A): Unit = {}
```

Comment:
why do we need this? I think I'm missing something
Also this test seems very similar to the one below apart from this and the test name.

Reply:
It is very similar, but this one does both a inlineable constant lookup and a method call. So in theory it should always work (as you can see by test `java interface method argument is direct` which has no version checks. However in practice it doesn't - in this test we need a version check of `2.12.0`. Hence there is some sort of issue which got resolved in 2.12.0 which made the checker now be able to detect this dependency.

The test below meanwhile only has an inlined constant lookup and hence needs to be 2.12.4+ for the fix.